### PR TITLE
Balder/remove v6 and v7 rebased 1

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli PR

### CODE REVIEW

**Review by vekterli — APPROVED** [review]
:+1: although this touches things that are _technically_ part of the public API, it's all stuff "implicitly" deprecated many years ago that should never have been public in the first place. Have done some sampling on internal code repos and found no references to the ancient code, so we're making a pragmatic assumption that no one is actually using this stuff.

File: document/src/main/java/com/yahoo/document/Field.java [hunk 1]
```diff
@@ -196,17 +177,13 @@ public void setDataType(DataType type) {
 
     /** Returns the numeric ID used to represent this field when serialized */
     public final int getId(int version) {
-        return (version > 6) ? getId() : getIdV6();
+        return getId();
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply document/src/main/java/com/yahoo/document/Field.java]: Deprecated and removed own use. Will drop in on vespa 8. This is a bit more public than the other api's touched.

File: document/src/test/java/com/yahoo/document/DocumentUpdateTestCase.java [hunk 2]
```diff
@@ -302,7 +303,7 @@ public void testSerialize() {
         docUp.addFieldUpdate(addMultiList);
 
         GrowableByteBuffer buf = new GrowableByteBuffer();
-        docUp.serialize(DocumentSerializerFactory.create42(buf));
+        docUp.serialize(DocumentSerializerFactory.create6(buf));
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply document/src/test/java/com/yahoo/document/DocumentUpdateTestCase.java]: I replace 42 serialize with 6 and 4.2 deserialize with HEAD to at least verify that new code can handle old data.

File: document/src/test/java/com/yahoo/document/DocumentUpdateTestCase.java [hunk 3]
```diff
@@ -403,7 +404,7 @@ public void testGenerateSerializedFile() throws IOException {
         upd.addFieldUpdate(serAdd);
 
         GrowableByteBuffer buf = new GrowableByteBuffer(100, 2.0f);
-        upd.serialize(DocumentSerializerFactory.create42(buf));
+        upd.serialize(DocumentSerializerFactory.create6(buf));
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply document/src/test/java/com/yahoo/document/DocumentUpdateTestCase.java]: Same resolution as above.

File: document/src/test/java/com/yahoo/document/serialization/SerializationHelperTestCase.java [hunk 4]
```diff
@@ -40,7 +38,7 @@ public void testGetNullTerminatedString() {
 
         assertTrue(data.position() == 0);
 
-        Utf8Array thisIsATestAgain = VespaDocumentDeserializer42.parseNullTerminatedString(data.getBuf().getByteBuffer(), 15);
+        Utf8Array thisIsATestAgain = VespaDocumentDeserializer6.parseNullTerminatedString(data.getBuf().getByteBuffer(), 15);
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply document/src/test/java/com/yahoo/document/serialization/SerializationHelperTestCase.java]: As above

File: document/src/test/java/com/yahoo/document/serialization/SerializeAnnotationsTestCase.java [hunk 5]
```diff
@@ -191,7 +191,7 @@ private static ByteBuffer writeFile(StringFieldValue value, String fileName) thr
         fileName = PATH + fileName;
 
         //serialize our tree to buffer
-        VespaDocumentSerializer42 serializer = new VespaDocumentSerializer42();
+        DocumentSerializer serializer = DocumentSerializerFactory.create6();
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply document/src/test/java/com/yahoo/document/serialization/SerializeAnnotationsTestCase.java]: As above

File: document/src/vespa/document/serialization/vespadocumentserializer.cpp [hunk 6]
```diff
@@ -156,87 +162,91 @@ void VespaDocumentSerializer::write(const Document &value,
...
+VespaDocumentSerializer::write(const AnnotationReferenceFieldValue &value)
 {
     putInt1_2_4Bytes(_stream, value.getAnnotationIndex());
 }
 
 void VespaDocumentSerializer::write(const ArrayFieldValue &value) {
     putInt1_2_4Bytes(_stream, value.size());
-    for (size_t i(0), m(value.size()); i < m; ++i) {
-        value[i].accept(*this);
+    for (auto & item : value) {
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply document/src/vespa/document/serialization/vespadocumentserializer.cpp]: Fixed

File: document/src/vespa/document/serialization/vespadocumentserializer.cpp [hunk 7]
```diff
@@ -391,8 +403,9 @@ void VespaDocumentSerializer::write(const StructFieldValue &value,
     _stream.write(toSerialize.c_str(), toSerialize.size());
 }
 
-void VespaDocumentSerializer::write(const WeightedSetFieldValue &value) {
-    const WeightedSetDataType *type = static_cast<const WeightedSetDataType *>(value.getDataType());
+void
+VespaDocumentSerializer::write(const WeightedSetFieldValue &value) {
+   auto type = static_cast<const WeightedSetDataType *>(value.getDataType());
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply document/src/vespa/document/serialization/vespadocumentserializer.cpp]: Fixed

**Pull request merged**